project-koku/koku
# [COST-2232] Discrepancies when using time_scope parameters

Fixes [COST-2232](https://issues.redhat.com/browse/COST-2232)

Changes proposed in this PR:
?filter[resolution]=monthly&filter[time_scope_value]=-3 Use to return time_scope_units day
```
 {
    "meta": {
        "count": 2,
        "filter": {
            "resolution": "monthly",
            "time_scope_value": "-3",
            "time_scope_units": "day"
        },
```
Now it will properly return month
```
    "meta": {
        "count": 10,
        "filter": {
            "resolution": "monthly",
            "time_scope_value": "-3",
            "enabled": true,
            "time_scope_units": "month"
```

----
Make sure you have RETAIN_NUM_MONTHS = 4 set in your env.
Setting up data generation for both parts of testing 
1. In the iqe-cost-managment edit the file `iqe_cost_management/data/aws/aws_static_report_basic.yml` by changing every start date to start_date: 2022-04-01.
2.  Add a breakpoint to test_api_aws_ingest_single_source
`@pytest.mark.cost_ingest_source
@pytest.mark.cost_smoke
@pytest.mark.cost_required
@pytest.mark.stage_api
@pytest.mark.stage_ui
@pytest.mark.outage
def test_api_aws_ingest_single_source(cost_aws_source_static):
    """Test to ingest aws single source

    metadata:
    requirements: cost_onboarding, cost_provider_management, cost_aws_support

    """
    breakpoint()
    assert cost_aws_source_static`
3. Continue to the testing instructions. 
---
Testing Instructions for the first part.
1. Make docker-up with this branch 
2. Run this test command in your test env `ENV_FOR_DYNACONF=local iqe tests plugin cost_management -k "test_api_aws_ingest_single_source" -vv --pdb`
3. The test will create a source that has data going back 3 months, once the test has hit the breakpoint move onto the next step.
4. Hit this endpoint http://localhost:8000/api/cost-management/v1/reports/aws/costs/?filter[resolution]=monthly&filter[time_scope_value]=-3
Expect to see this result:
```
 {
    "meta": {
        "filter": {
            "resolution": "monthly",
            "time_scope_value": "-3",
            "time_scope_units": "month"
        },
```
5. Hit this endpoint http://localhost:8000/api/cost-management/v1/tags/aws/?filter[resolution]=monthly&filter[time_scope_value]=-3
6. The endpoint will successfully return data 
  
---
